lib/tasks/missing_payments.rake

```diff
@@ -0,0 +1,56 @@
+# Find gaps in the sequence of payment ids.
+# If there are gaps then see if there is a log entry with a payment result for
+# the now lost payment. If there are some then you probably want to follow up
+# with the affected enterprise and see if customers need to be refunded.
+#
+# ## Usage
+
+# Report of the last 35 days:
+#   rake ofn:missing_payments[35]
+namespace :ofn do
+  desc 'Find payments that got lost'
+  task :missing_payments, [:days] => :environment do |_task_, args|
+    days = args[:days].andand.to_i || 7
+    payments_sequence = Spree::Payment.where("created_at > ?", days.days.ago).order(:id).pluck(:id)
+    payments_range = (payments_sequence.first..payments_sequence.last).to_a
+    missing_payment_ids = payments_range - payments_sequence
+    puts "Gaps in the payments sequence: #{missing_payment_ids.count}"
```

Comment:
We should avoid using `puts` and use `Rails.logger` instead. The Rails logging is better suited for these.

Reply:
I don't see this as log message. It's the defined output of the program. It's like `ls` is listing files. I don't think it should use a logger for that.